extensions/WSLExtension/Models/WslInstallDistributionOperation.cs

```diff
@@ -93,11 +105,102 @@ public IAsyncOperation<CreateComputeSystemResult> StartAsync()
             }
             catch (Exception ex)
             {
-                _log.Error(ex, $"Unable to install {_definition.FriendlyName} due to exception");
+                _log.Error(ex, $"Unable to install and register {_definition.FriendlyName} due to exception");
                 var errorMsg = _stringResource.GetLocalized("WSLInstallationFailedWithException", _definition.FriendlyName, ex.Message);
                 return new CreateComputeSystemResult(ex, errorMsg, ex.Message);
             }
-        }).AsAsyncOperation();
+            finally
+            {
+                _wslManager.WslInstallationEventHandler -= OnInstallChanged;
+            }
+        });
+    }
+
+    private void StatusUpdateCallback(string progressText)
+    {
+        StatusUpdateCallback(progressText, IndeterminateProgressPercentage);
+    }
+
+    private void StatusUpdateCallback(string progressText, uint progressPercent)
+    {
+        try
+        {
+            Progress?.Invoke(this, new CreateComputeSystemProgressEventArgs(progressText, progressPercent));
+        }
+        catch (Exception ex)
+        {
+            _log.Error(ex, "Failed to provide progress back to Dev Home");
+        }
+    }
+
+    private void OnInstallChanged(object? sender, AppInstallItem args)
+    {
+        var packageName = _definition.FriendlyName;
+
+        if (!_definition.StoreAppId.Equals(args.ProductId, StringComparison.OrdinalIgnoreCase))
+        {
+            // If we're not downloading/installing the wsl distribution with the provided productId
+            // then check if the Linux kernel package is being downloaded/installed.
+            if (!WslKernelPackageStoreId.Equals(args.ProductId, StringComparison.OrdinalIgnoreCase))
+            {
+                // The AppInstallItem isn't the selected distribution nor is it the kernel package.
+                return;
+            }
+
+            packageName = _stringResource.GetLocalized("WslKernelPackageName");
+        }
+
+        var status = args.GetCurrentStatus();
+        var itemInstallState = status.InstallState;
+        var progressText = GetLocalizedString("AppInstallPending", packageName);
+        var progressPercent = IndeterminateProgressPercentage;
+
+        switch (itemInstallState)

```

Comment:
Are these states common to all installations? If yes, then they should be part of the general package installation class.

Reply:
These states come from https://learn.microsoft.com/en-us/uwp/api/windows.applicationmodel.store.preview.installcontrol.appinstallstate?view=winrt-26100. The `MicrosoftStoreService` in the Dev Home services project has a  `StatusChanged`   event here: https://github.com/microsoft/devhome/blob/87d21866b197db64b1d779aeba34b9ecae4c6f7d/services/DevHome.Services.Core/Services/MicrosoftStoreService.cs#L19 that we subscribe to. That enum value is located in the `AppInstallManagerItemEventArgs` of that event.

Reply:
I meant moving this logic to a more general class from this file.

Reply:
Spoke offline, won't fix this one for now since we're the only place in Dev Home using the MicrosoftStoreService to send information to the user. In the future we can move this out to a common place if need be.